app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,45 @@
 class PostsController < ApplicationController
-  # index action displaying a list of posts belonging to a specific user
-  # first retrieves the user record base on user_id
-  # next fetches all posts associated with that user using @user.posts
+  before_action :set_user, only: %i[index create]
+  before_action :set_post, only: %i[show like]
 
   def index
-    @user = User.find(params[:user_id])
     @posts = @user.posts
   end
 
   def show
+    @post = Post.find(params[:id]) # Find the post by its id
+    @comment = Comment.new
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = @user.posts.build(post_params)
+    if @post.save
+      redirect_to post_path(@post), notice: 'Post created successfully.'
```

Comment:
- [ ] A post is not correctly redirected after saving to the database. Please check screen cast above to confirm this.